src/test/ui/save-analysis/issue-89066.rs

```diff
@@ -0,0 +1,31 @@
+// compile-flags: -Zsave-analysis
+
+// Check that this does not ICE.
+// Stolen from src/test/ui/inference/infer-arg-test.rs
+
+#![feature(generic_arg_infer)]
+
+struct All<'a, T, const N: usize> {
+  v: &'a T,
+}
+
+struct BadInfer<_>;
+//~^ ERROR expected identifier
+//~| ERROR parameter `_` is never used
+
+fn all_fn<'a, T, const N: usize>() {}
+
+fn bad_infer_fn<_>() {}
+//~^ ERROR expected identifier
+
+
+fn main() {
+  let a: All<_, _, _>;
+  //~^ ERROR this struct takes 2 generic arguments but 3 generic arguments were supplied
+  // FIXME(bgr360): this error does not occur in src/test/ui/inference/infer-arg-test.rs
```

Comment:
Not sure why this is now erroring. Possible side effect of `save-analysis`? Should we open up another issue for this?

Reply:
This is indeed a side-effect of `-Zsave-analysis`. Without that flag, we're halting compilation due to the previous errors. When it's included, the `DumpVisitor` in `rustc_save_analysis` is calling the `typeck` query leading to this error.

Reply:
Cool, thx. I'll get rid of the FIXME